src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -384,6 +390,12 @@ def set(self, hostname: str, labels: List[str]):
         """
         orch = OrchClient.instance()
         host = get_host(hostname)
+        # only allow List[str] type for labels
+        if not isinstance(labels, list):
+            raise DashboardException(
+                msg='Operation exception. Please check the response body for details.',
```

Comment:
I think the message can be more explicit (i.e. "Expected list of labels. Please check API documentation.").

Reply:
I just copied the message from doc itself https://docs.ceph.com/en/latest/mgr/ceph_api/#put--api-host-hostname for 400 error

Reply:
That is the generic message. IMHO giving more details when possible is helpful to the end user. Example:
https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/services/rgw_client.py#L655

Reply:
Yes, agree. Done!